packages/core/src/request.ts

```diff
@@ -37,10 +37,20 @@ export function eventToSentryRequest(event: Event, api: API): SentryRequest {
   if (useEnvelope) {
     const envelopeHeaders = JSON.stringify({
       event_id: event.event_id,
+
       // We need to add * 1000 since we divide it by 1000 by default but JS works with ms precision
       // The reason we use timestampWithMs here is that all clocks across the SDK use the same clock
       sent_at: new Date().toISOString(),
+
+      // trace context for dynamic sampling on relay
+      trace: {
+        trace_id: event.contexts?.trace?.trace_id,
+        // TODO: any reason we can't change this property to be called publicKey, since that's what it is?
```

Comment:
I guess relay is expecting `public_key`?

Reply:
No, sorry. I was unclear. I meant this: https://github.com/getsentry/sentry-javascript/compare/kmclb-user-in-dsn-is-public-key.